pkg/autostart/autostart.go

```diff
@@ -0,0 +1,137 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		// Early return if error is unexpected
+		if !errors.Is(err, os.ErrExist) && !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Unit file exists, return nil
+		if errors.Is(err, os.ErrExist) {
```

Comment:
Shouldn't we overwrite ??

If for some reason, there is a template data change, we might need to overwrite and reinstall the service right ?

Reply:
I dont get, could you explain?

Reply:
I think @balajiv113 means that we should simply overwrite the file, even if it already exists.

But I believe `os.WriteFile` will already do that, so the only way we could get `os.ErrExist` would be if the file exists, but we don't have permission to write.

In which case I think an error should be reported because the content may now not be what we expect it to be.

Although this case should be rather obscure: (1) someone must have intentionally removed the write permission, and (2) something like the Lima install location must have changed, so the previous content would no longer be valid.

I would just remove all the special logic for `os.ErrExist` and return it like any other error.